src/services/download-file.ts

```diff
@@ -91,6 +91,34 @@ export default async function downloadFile(blob: Blob, fileName: string) {
   }
 }
 
+export default async function downloadAttachment(attachment: AttachmentValid) {
+  //
+  if (attachment.type) {
+    return await downloadFile(attachment.data, attachment.fileName)
+  }
+  let idToken
+  if (attachment.isPrivate) {
+    idToken = await authService.getIdToken()
+  }
+  const response = await fetch(
+    attachment.url,
+    idToken
+      ? {
+          headers: {
+            Authorization: `Bearer ${idToken}`,
+          },
+        }
+      : undefined,
+  )
+  if (!response.ok) {
+    throw new Error(
+      `Unable to download file. HTTP Status Code: ${response.status}`,
+    )
+  }
+  const blob = await response.blob()
```

Comment:
All of this needs to be wrapped in that try catch. Downloading files should not throw an errors.